src/activity_handlers.cpp

```diff
@@ -598,23 +598,23 @@ void activity_handlers::forage_finish( player_activity *act, player *p )
     bool found_something = false;
 
     items_location loc;
-    ter_id next_ter = t_null; //Just to have a default for compilers' sake
+    ter_str_id next_ter = NULL_ID; //Just to have a default for compilers' sake
```

Comment:
You can remove the explicit initialization (and the comment), the default constructor of `string_id` prevents the compiler warning and value is supposed to be overridden on all code paths anyway.